share/functions/__fish_inc_dec_number_under_cursor.fish

```diff
@@ -0,0 +1,90 @@
+# Usage:
+    # This function can be bound to key shortcuts (like Ctrl+A for increase and Ctrl+X for decrease) in fish shell to modify numbers quickly under the cursor.
+    # Key bindings for increasing and decreasing numbers in insert/vi mode mode.
+    # bind -M insert \ca '__fish_inc_dec_number_under_cursor increase'
+    # bind -M insert \cx '__fish_inc_dec_number_under_cursor decrease'
+
+function __fish_inc_dec_number_under_cursor
```

Comment:
this is tricky. Let's try to find ways to simplify it. Unfortunately fish is not really made to do this well.
I expect that we can make major simplifications and get rid of almost all the sed/cut/grep usage.